frontend/packages/topology/console-extensions.json

```diff
@@ -5,5 +5,138 @@
       "scope": "devconsole",
       "reducer": { "$codeRef": "reduxReducer" }
     }
+  },
+  {
+    "type": "console.topology/details/tab",
+    "properties": {
+      "id": "topology-side-bar-tab-details",
+      "label": "%topology~Details%"
+    }
+  },
+  {
+    "type": "console.topology/details/tab",
+    "properties": {
+      "id": "topology-side-bar-tab-resource",
+      "label": "%topology~Resources%"
+    }
+  },
+  {
+    "type": "console.topology/details/tab-section",
+    "properties": {
+      "id": "topology-tab-section-deployment-config",
+      "tab": "topology-side-bar-tab-details",
+      "section": { "$codeRef": "workload.getDeploymentConfigSideBarDetails" }
+    }
```

Comment:
I feel like there should be a default "details" contribution for all elements. It would resolve the resource and display the standard details. if there is a custom contribution it can override the section by providing a higher priority along with using the same `id`.
It's kind of like how all our details pages today have a standard page that can be overridden.

I think all the cases you've added need the customization however side bars for the service binding will need a default contribution without having to actually create a plugin to contribute one directly.